commit 4c366df74acee4dfc57c3d0ab7ac8b080d9eaea7
Author: Leonardo Alminana <leonardo@calyptia.com>
Date:   Mon Oct 24 11:57:32 2022 +0200

    storage: modified the log_cb prototype to match the one expected by cio
    
    Signed-off-by: Leonardo Alminana <leonardo@calyptia.com>

diff --git a/src/flb_storage.c b/src/flb_storage.c
index 44d9b2e86e4422cd..004df4696a7235a7 100644
--- a/src/flb_storage.c
+++ b/src/flb_storage.c
@@ -360,8 +360,8 @@ static void print_storage_info(struct flb_config *ctx, struct cio_ctx *cio)
     }
 }
 
-static void log_cb(void *ctx, int level, const char *file, int line,
-                   const char *str)
+static int log_cb(struct cio_ctx *ctx, int level, const char *file, int line,
+                  char *str)
 {
     if (level == CIO_LOG_ERROR) {
         flb_error("[storage] %s", str);
@@ -375,6 +375,8 @@ static void log_cb(void *ctx, int level, const char *file, int line,
     else if (level == CIO_LOG_DEBUG) {
         flb_debug("[storage] %s", str);
     }
+
+    return 0;
 }
 
 int flb_storage_input_create(struct cio_ctx *cio,
